app/database.py

```diff
@@ -25,6 +25,24 @@ def getChallenge(post_id):
     challenge = db.session.query(Challenge).filter_by(id=post_id).first()
     return challenge
 
+def getChallengesByTitle(search):
+    challenges = db.session.query(Challenge).filter(Challenge.title.icontains(search)).all()
+    return challenges
+
+def getChallengesByCreator(search):
```

Comment:
Are this and by title even necessary?

Reply:
Additionally, consider if having search be a seperate route is ideal - redirecting to index with something like a query
?search="searchstring"